SampleOrigin/wwwroot/Index.html

```diff
@@ -0,0 +1,49 @@
+﻿<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8" />
+    <script type="text/javascript">
+        
+        function createCORSRequest(method, url) {          
+            var request = new XMLHttpRequest();
+            request.open(method, url, true);
+            return request;
+        }
+
+        // Make the CORS request.
+        function makeCORSRequest(method) {
+
+            // Destination server with CORS enabled.
+            var url = 'http://destination.sample.com/api';
+            var request = createCORSRequest(method , url);
+
+            if (!request) {
+                alert('CORS not supported');
+                return;
+            }
+            
+            // Response handlers.
+            request.onload = function () {
+                var text = request.responseText;
+                alert('Response from CORS '+method+' request to ' + url + ': ' + text);
+            };
+
+            request.onerror = function () {
+                alert('There was an error making the request for method ' + method);
+            };
+
+            request.send();
+        }
+
+    </script>
+    <title></title>
+</head>
+<body>
+    <form method="post">
+            <input id="clickMe" type="button" value="CORS Request" onclick="makeCORSRequest('GET');" />
```

Comment:
This is a bigger ask but it would be nice if the details of the CORS request was dynamic in nature. I say this because setting up a repro for CORS (whenever a user says that some scenario is not working) is a pain so if we could add a bit of dynamic nature to this sample it would be very helpful for us, internally, to quickly validate issues filed by users. By dynamic, I mean for example, selecting the http method via a drop down or posting data of different content-types etc. Please check with @Eilon regarding this.